plugins/modules/event.py

```diff
@@ -0,0 +1,279 @@
+        type: str
+      state:
+        description: The state of the check.
+        choices: [ "passing", "failing", "flapping" ]
+        type: str
+      status:
+        description: Exit status code produced by the check.
+        choices: [ "ok", "warning", "critical", "unknown" ]
+        type: str
+      total_state_change:
+        description: The total state change percentage for the check's history.
+        type: int
+  metric_attributes:
+    type: dict
+    description:
+      - Metric attributes. Find out more at
+        U(https://docs.sensu.io/sensu-go/latest/reference/events/#metric-attributes)
+    suboptions:
+      handlers:
+        description:
+          - An array of Sensu handlers to use for events created by the check.
+            Each array item must be a string.
+        type: list
+        elements: str
+      points:
+        description: Metric data points including a name, timestamp, value, and tags.
+        type: list
+        elements: dict
+'''
+
+EXAMPLES = '''
+- name: Create event
+  event:
+    auth:
+      url: http://localhost:8080
+    entity: awesome_entity
+    check: awesome_check
+    check_attributes:
+      duration: 1.945
+      executed: 1522100915
+      history:
+        - executed: 1552505193
+          status: 1
+      issued: 1552506034
+      last_ok: 1552506033
+      output: '10'
+      state: 'passing'
+      status: 'ok'
+      total_state_change: 0
+    metric_attributes:
+      handlers:
+        - handler1
+        - handler2
+      points:
+        - name: "sensu-go-sandbox.curl_timings.time_total"
+          tags:
+            - name: "response_time_in_ms"
+              value: 101
+          timestamp: 1552506033
+          value: 0.005
+        - name: "sensu-go-sandbox.curl_timings.time_namelookup"
+          tags:
+            - name: "namelookup_time_in_ms"
+              value: 57
+          timestamp: 1552506033
+          value: 0.004
+'''
+
+RETURN = '''
+object:
+  description: object representing Sensu event
+  returned: always
+  type: dict
+'''
+
+from ansible.module_utils.basic import AnsibleModule
+
+from ansible_collections.sensu.sensu_go.plugins.module_utils import (
+    arguments, errors, utils
+)
+
+STATUS_MAP = {
+    'ok': 0,
+    'warning': 1,
+    'critical': 2,
+    'unknown': 3,
+}
+
+
+def get_check(client, check):
+    check_path = '/checks/{0}'.format(check)
+    resp = client.get(check_path)
+    if resp.status != 200:
+        raise errors.SyncError("Check with name '{0}' does not exist on remote.".format(check))
+    return resp.json
+
+
+def get_entity(client, entity):
+    entity_path = '/entities/{0}'.format(entity)
+    resp = client.get(entity_path)
+    if resp.status != 200:
+        raise errors.SyncError("Entity with name '{0}' does not exist on remote.".format(entity))
+    return resp.json
+
+
+def _update_payload_with_metric_attributes(payload, metric_attributes):
+    if not metric_attributes:
+        return None
```

Comment:
Just a tiny nitpick: if the function returns `void`, it should return `void` in all branches. Now, because python does not know how to return `void`, it returns `None` in such cases. This makes this line valid, but it would be considered more correct to just call `return` with no parameters, since this gives another indication that the return value of this function is not meant to be used.